### DISCUSSION TITLE
issue with static files

### DISCUSSION THREAD

**Discussion by remior** [netbox-community/netbox#11515] [General]
hello i m using an old release v2.10.2

i configure everything but i have an issue with static file for which i receive randomly 404 on nginx

`10.193.71.39 - - [16/Jan/2023:18:00:10 +0100] "GET / HTTP/1.1" 200 6091 "http://10.192.230.14/media-failure/?filename=css/base.css" "Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:108.0) Gecko/20100101 Firefox/108.0"
10.193.71.39 - - [16/Jan/2023:18:00:10 +0100] "GET /static/css/base.css?v2.10.2 HTTP/1.1" 200 8528 "http://10.192.230.14/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:108.0) Gecko/20100101 Firefox/108.0"
10.193.71.39 - - [16/Jan/2023:18:00:10 +0100] "GET /static/jquery/jquery-3.5.1.min.js HTTP/1.1" 200 89476 "http://10.192.230.14/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:108.0) Gecko/20100101 Firefox/108.0"
10.193.71.39 - - [16/Jan/2023:18:00:10 +0100] "GET /static/bootstrap-3.4.1-dist/js/bootstrap.min.js HTTP/1.1" 200 39680 "http://10.192.230.14/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:108.0) Gecko/20100101 Firefox/108.0"
10.193.71.39 - - [16/Jan/2023:18:00:10 +0100] "GET /static/select2-4.0.13/dist/js/select2.min.js HTTP/1.1" 200 70891 "http://10.192.230.14/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:108.0) Gecko/20100101 Firefox/108.0"
10.193.71.39 - - [16/Jan/2023:18:00:10 +0100] "GET /static/clipboard.js/clipboard-2.0.6.min.js HTTP/1.1" 200 10454 "http://10.192.230.14/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:108.0) Gecko/20100101 Firefox/108.0"
10.193.71.39 - - [16/Jan/2023:18:00:10 +0100] "GET /static/flatpickr-4.6.3/flatpickr.min.js HTTP/1.1" 200 48518 "http://10.192.230.14/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:108.0) Gecko/20100101 Firefox/108.0"
10.193.71.39 - - [16/Jan/2023:18:00:10 +0100] "GET /static/js/forms.js?v2.10.2 HTTP/1.1" 200 18595 "http://10.192.230.14/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:108.0) Gecko/20100101 Firefox/108.0"
10.193.71.39 - - [16/Jan/2023:18:00:10 +0100] "GET /static/jquery-ui-1.12.1/jquery-ui.min.js HTTP/1.1" 200 139907 "http://10.192.230.14/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:108.0) Gecko/20100101 Firefox/108.0"
10.193.71.39 - - [16/Jan/2023:18:00:10 +0100] "GET /media-failure/?filename=materialdesignicons-5.4.55/css/materialdesignicons.min.css HTTP/1.1" 200 843 "http://10.192.230.14/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:108.0) Gecko/20100101 Firefox/108.0"
`

sometimes it happens on differents files. where can i look to understand this issue? (when i make a curl request i m hable to reach the file missing...

**Comment by candlerb** [comment]
Sorry but I don't see a 404 in any of those logs.  They all show "200".

**Reply by remior** [reply]
i got : "GET /media-failure/?filename=materialdesignicons-5.4.55/css/materialdesignicons.min.css HTTP/1.1" 200 843 "http://10.192.230.14/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:108.0) Gecko/20100101 Firefox/108.0"

**Reply by remior** [reply]
it 200 but error :)

**Reply by remior** [reply]
and if i launch again i got another error on another file

10.192.149.91 - - [17/Jan/2023:10:40:58 +0100] "GET / HTTP/1.1" 200 6091 "-" "Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:108.0) Gecko/20100101 Firefox/108.0"
10.192.149.91 - - [17/Jan/2023:10:40:58 +0100] "GET /static/bootstrap-3.4.1-dist/css/bootstrap.min.css HTTP/1.1" 200 77922 "http://10.192.230.14/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:108.0) Gecko/20100101 Firefox/108.0"
10.192.149.91 - - [17/Jan/2023:10:40:58 +0100] "**GET /media-failure/?filename=bootstrap-3.4.1-dist/css/bootstrap.min.css** HTTP/1.1" 200 838 "http://10.192.230.14/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:108.0) Gecko/20100101 Firefox/108.0"

**Reply by candlerb** [reply]
But you haven't shown _any_ error. An error *where*? In the browser? In some other log file? What makes you say it's a "404" error specifically?

If it's in the browser, show developer console logs. If it's in an upstream proxy, then show logs from the upstream proxy.

Have you got some anti-virus or ad-blocking software installed in your browser? Can you try again with a clean browser?

I do notice from your logs that the front-end has made a fetch to `/media-failure/...` and I know it's the browser that's fetching that[^1].  However, since the webserver has clearly returned the requested file (with 200 result code and a plausible content length) it must be further towards the browser that the problem is occurring; this is not a problem with Netbox.

-----
[^1] The redirects come from resource `onerror` handlers in `netbox/templates/base.html`:

```
<script src="{% static 'bootstrap-3.4.1-dist/js/bootstrap.min.js' %}"
        onerror="window.location='{% url 'media_failure' %}?filename=bootstrap-3.4.1-dist/js/bootstrap.min.js'"></script>
```

So clearly, this is an error is detected at the browser side.  If you want any clues, you'll need to look at browser-side logs (developer console and network tab)

`onerror` could mean that the script wasn't downloaded successfully (the webserver says it was, but it might have been dropped by some intervening proxy). It could also mean that the script couldn't be parsed (e.g. you have some ancient browser that doesn't understand modern Javascript). It might be some browser security policy issue, e.g. trying to fetch Javascript over HTTP when the main page is accessed via HTTPS - the browser console would tell you if this is the case. Or you might have some browser plugin or other filtering tool which is rejecting this content.

For reference, in Netbox v2.10.2, the backend handler is in `netbox/netbox/urls.py` and `netbox/netbox/views/__init__.py`

```
    path('media-failure/', StaticMediaFailureView.as_view(), name='media_failure'),
```

```
class StaticMediaFailureView(View):
    """
    Display a user-friendly error message with troubleshooting tips when a static media file fails to load.
    """
    def get(self, request):
        return render(request, 'media_failure.html', {
            'filename': request.GET.get('filename')
        })
```

**Reply by remior** [reply]
thanks for all these clear explanations. i ll come back to you after performing some test with different browsers

**Comment by remior** [comment]
it was an issue on my nginx config forgott to add include mime.types;
